src/intersection/first-solution/intersection.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Creates an array of values that are in both the first and the second arrays.
+ *
+ * @param {Array} [firstArr=[]] - First array to be compared for repeated values.
+ * @param {Array} [secondArr=[]] - Second array to be compared for repeated values.
+ * @returns {Array} Returns the new array with only intersected values.
+ */
+
+export const arrayIntersection = (firstArr = [], secondArr = []) => {
+  let newArrayIntersected = [];
+  // Test if the arguments are arrays and make sure that arguments are passed
+  if (!Array.isArray(firstArr) || !Array.isArray(secondArr)) {
+    throw new TypeError('The arguments must be arrays');
```

Comment:
Just to point the importance of @throw in JSDoc  that all calls of this function will required to be in try-catch blocs otherwise when you throw error the program will stop:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw 

"The throw statement throws a user-defined exception. Execution of the current function will stop (the statements after throw won't be executed), and control will be passed to the first catch block in the call stack. If no catch block exists among caller functions, the program will terminate."

Reply:
That is a documentation explicit to explain how the throw works, and in mine I'm explaining how intersection works